native/metrodroid/metrodroid.xcodeproj/project.pbxproj

```diff
@@ -0,0 +1,891 @@
+					};
+				};
+			};
+			buildConfigurationList = E7B1DFBB22CCCE680061481F /* Build configuration list for PBXProject "metrodroid" */;
+			compatibilityVersion = "Xcode 9.3";
+			developmentRegion = en;
+			hasScannedForEncodings = 0;
+			knownRegions = (
+				en,
+				Base,
+				ru,
+				es,
+				fr,
+				fi,
+				he,
+				id,
+				ja,
+				nl,
+				tr,
+				kab,
+				nb,
+				"zh-Hans",
+				de,
+				it,
+				el,
+			);
+			mainGroup = E7B1DFB722CCCE680061481F;
+			packageReferences = (
+				9CC7BA2622F6582700F45148 /* XCRemoteSwiftPackageReference "ZIPFoundation" */,
+			);
+			productRefGroup = E7B1DFC122CCCE680061481F /* Products */;
+			projectDirPath = "";
+			projectRoot = "";
+			targets = (
+				E7B1DFBF22CCCE680061481F /* metrodroid */,
+				E7B1DFD322CCCE6B0061481F /* metrodroidTests */,
+				E7B1DFDE22CCCE6B0061481F /* metrodroidUITests */,
+			);
+		};
+/* End PBXProject section */
+
+/* Begin PBXResourcesBuildPhase section */
+		E7B1DFBE22CCCE680061481F /* Resources */ = {
+			isa = PBXResourcesBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+				E7BE9C9222CD6DEE00E6496B /* NOTICE.AOSP.txt in Resources */,
+				9C5352B422F2750D00674952 /* iOSDeviceModelMapping.plist in Resources */,
+				E7BE9C8222CD24E600E6496B /* Settings.bundle in Resources */,
+				9CC7BA3322F7CD7A00F45148 /* Logos-NOTICE.txt in Resources */,
+				E7BE9C9022CD6DEE00E6496B /* NOTICE.protobuf.txt in Resources */,
+				9CEA951922DD14120072C513 /* mdst in Resources */,
+				E7B1DFCB22CCCE6B0061481F /* Assets.xcassets in Resources */,
+				E7B1DFC922CCCE680061481F /* Main.storyboard in Resources */,
+				9CCEE1ED22E3D181007BF742 /* InfoPlist.strings in Resources */,
+				E7BE9C9122CD6DEE00E6496B /* NOTICE.noto-emoji.txt in Resources */,
+				9CC7BA3522F7CD8600F45148 /* Metrodroid-NOTICE.txt in Resources */,
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+		};
+		E7B1DFD222CCCE6B0061481F /* Resources */ = {
+			isa = PBXResourcesBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+		};
+		E7B1DFDD22CCCE6B0061481F /* Resources */ = {
+			isa = PBXResourcesBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+		};
+/* End PBXResourcesBuildPhase section */
+
+/* Begin PBXShellScriptBuildPhase section */
+		E7C4D74B22CF055A00E89028 /* ShellScript */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "cd \"$SRCROOT/../../build/xcode-frameworks/\"\n./gradlew :proto:mainSharedLibrary\n./gradlew :proto:mainStaticLibrary\n./gradlew :generateLocalize :iOSLanguages :iOSMappedLanguages\n./gradlew :packForXCode -PXCODE_CONFIGURATION=${CONFIGURATION}\n";
```

Comment:
`cd "$SRCROOT/../../build/xcode-frameworks/"` should be `cd "$SRCROOT/../../"` instead

Reply:
Ah, I see, this is generated in `packForXCode`, in order to use the same version of Java as Android Studio is set to use.  But unless you've run a build of the Android version already, this task never gets triggered.

Reply:
The plot thickens -- `packForXCode` doesn't seem to be triggering on a regular Android build. When explicitly triggering this target, Android Studio uses the stable (rather than beta) version of Xcode:

```
> Task :linkMetrolibDebugFrameworkIOS
Undefined symbols for architecture x86_64:
  "_objc_opt_class", referenced from:
      -[GPBInt32Array dealloc] in libmain.a(GPBProtocolBuffers.o)
      -[GPBInt32Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBInt32Array description] in libmain.a(GPBProtocolBuffers.o)
      _GPBAutocreatedArrayModified in libmain.a(GPBProtocolBuffers.o)
      -[GPBUInt32Array dealloc] in libmain.a(GPBProtocolBuffers.o)
      -[GPBUInt32Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBUInt32Array description] in libmain.a(GPBProtocolBuffers.o)
      ...
  "_objc_opt_isKindOfClass", referenced from:
      -[GPBInt32Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBUInt32Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBInt64Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBUInt64Array isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBFloatArray isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBDoubleArray isEqual:] in libmain.a(GPBProtocolBuffers.o)
      -[GPBBoolArray isEqual:] in libmain.a(GPBProtocolBuffers.o)
      ...
ld: symbol(s) not found for architecture x86_64
e: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld invocation reported errors

> Task :linkMetrolibDebugFrameworkIOS FAILED
```

For comparison:

```
$ '/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java' -version
openjdk version "1.8.0_152-release"
OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
OpenJDK 64-Bit Server VM (build 25.152-b01, mixed mode)

$ java -version
openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)
```

I think e4087b9d resolves this, but on a system that doesn't have Java installed outside of Android Studio, this may do weird things.

Reply:
I think xcode pulls java in as well. In any case, I'm fine with either approach.